### PR TITLE
[Settings] Add adjust display refresh rate start/stop with VFR

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

Followup of #21372
Add new setting to allow resolution change during playback for VFR videos also when set Start/Stop

## Motivation and context

In previous PR https://github.com/xbmc/xbmc/pull/21372#issuecomment-1135788965 comment is explained a use case for VFR that now is handled only by set: "adjust display refresh rate" to "always"

being that Start/Stop setting now does exactly its function, will not works annymore with VFR

then if you play a VFR video you need to set "adjust display refresh rate" to "always" to allow resolution change during playback
but this setting however will change refreshrate also when you return to the mainmenu GUI while playing

so i added a new setting to allow VFR video change refreshrate with Start/Stop

## How has this been tested?

Played VFR video with various use cases of "adjust display refresh rate"

## What is the effect on users?

Allow start/stop with refresh rate change during playback for VFR videos, but without change refresh rate when you return to the GUI mainmenu during playback

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by da-anda** [comment]
@FernetMenta mind a comment if this fixes the issue you meant, or will this still cause issues due to the distinction between initial configure and reconfigure?

@CastagnaIT thanks for improving the comments on the labels. I noticed that the help text is not really that helpful with all these different options and no further explanation. It might be good to explain the to be expected behavior of all these options, what do you think? But let's first wait on Fernets comment if this fixes the issue he mentioned or if we need an entirely different approach to the VFR thing (or adust display refresh rate in general)

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -18732,12 +18732,20 @@ msgctxt "#36049"
...
 #: system/settings.xml
 msgctxt "#36050"
 msgid "On start"
 msgstr ""
 
-#empty strings from id 36051 to 36097
+#. Item list value of setting with label #170 "Adjust display refresh rate"
+#. VFR is abbreviation for: Variable Frame Rate
+#: system/settings.xml
+msgctxt "#36051"
```
[the remark this answers is not in the document]
Comment (CastagnaIT) [reply addons/resource.language.en_gb/resources/strings.po:18744]: An alternative could be "During playback", but also "Always" setting IMO is not so intuitive,
whats difference? could be an idea rename also "always" to something like "During playback and GUI menus"

**Comment by FernetMenta** [comment]
@da-anda I don't think this restores lost behaviour done by the previous changes. The assumption that video fps is known at start of playback is wrong for many cases. This is why VP has measured fps and then triggered the change. This behaviour is not related to VFR. With this PR things get even worse.

**Comment by CastagnaIT** [comment]
>  I don't think this restores lost behaviour done by the previous changes

you are spamming without testing things
i havent done the PR WITHOUT TESTING
the behaviour is exactly the original before my changes when you set ADJUST_REFRESHRATE_ON_STARTSTOP_VFR setting

otherwise you are saying that the original "adjust display refresh rate" setting was wrong implemented since its birth...

**Comment by CastagnaIT** [comment]
these are the differences:

### ORIGINAL Start/Stop behaviour
at playback start resolution change is triggered:
1 time: VP hints https://github.com/CastagnaIT/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L3594
1 time: with video data "STATE_CONFIGURING"
and all times fps change while in playback

### AFTER MY FIRST FIX PR Start/Stop behaviour
at playback start resolution change is triggered:
1 time: VP hints https://github.com/CastagnaIT/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L3594
1 time: with video data "STATE_CONFIGURING"

### WITH THIS PR: Start/Stop + "VFR"
at playback start resolution change is triggered:
1 time: VP hints https://github.com/CastagnaIT/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L3594
1 time: with video data "STATE_CONFIGURING"
and all times fps change while in playback

**Comment by CastagnaIT** [comment]
> The assumption that video fps is known at start of playback is wrong for many cases.

this example https://github.com/xbmc/xbmc/issues/21447 we dont get the fps from starts,
so we have other problems elsewhere if we do not get this information, other software are able to get it
then there are bugs not related to this

**Comment by fritsch** [comment]
Don't overshoot. Especially in live stream scenarios with interlaced content and very few time to parse ahead, you cannot assume that this is known.

The old design cared exactly for those unknowns by switching ones when known. The VFR content is in comparison to the named Livestreams far less existent, so might mean you did not break something working by accident, but you actually broke a sane mitigation by not giving the player this chance anymore.

As your usecase to fix is just VFR, hints might be a good idea here.

**Comment by thexai** [comment]
I totally agree with what fritsch and Fernet have mentioned.

Cases to be taken into account:

- CFR videos with fps know previous to start playback (.MKV's etc.)
- CFR with fps = 0 (unknown) at beginning --> This is broken now and is more important functionality that VFR (i.e .ts streams)
- VFR videos --> honestly, I don't know if it's broken or not because I'm not interested in this functionality and I've never played this kind of videos.

**Comment by FernetMenta** [comment]
> so if we want handle these weird cases of fps 0

Those are no weird cases. Look at mpegts with h264 for example. There is nothing in the specification that gives you the frame rate. There is a long history of CDVDPlayerVideo::CalcFrameRate(). You should have got familiar with this before changing behaviour that breaks a lot.

**Pull request closed without merging**